### PR TITLE
[WIP] making id verification an inline process

### PR DESCRIPTION
**Pull request by fzhao99**
## Related Issue or Background Info
Changing verify organizations to be an inline action rather than one

## Changes Proposed

- Moved the verify button inline and removed the global ID verify checkbox/verify flow
- Combined edit/verification into one modal per design feedback that highlighted having editing/verification in two separate columns was redundant
- Added a separate "Save details" CTA that allows for the support team to save without editing

## Additional Information

- Added functionality that checks for any changes not saved prior to submit on the event that someone fills out a form without saving then submits it
- May want some better user feedback for "successfully edited" beyond just the screen changing with the new info
- Ripped out the code that allowed for bulk verification in favor of the individual action in favor of individualized version.

## Screenshots / Demos

https://user-images.githubusercontent.com/29645040/145445932-a15e4700-24c1-4922-be81-5c4a00a4ee40.mov

## Checklist for Author and Reviewer
- [ ] Looking at implementation within the `PendingOrganizations.tsx`, `PendingOrganizationsContainer.tsx`, and `ConfirmOrgVerificationModal.tsx` files
- [ ] Checking the updated tests within `PendingOrganizationsContainer.test.tsx` to see if I've missed anything/wrote something badly
- [ ] Give feedback to Bob about anything he didn't do correctly around the PR process (first real one to a code base this large! :D)

### UI
- [x ] Any changes to the UI/UX are approved by design 
- [x ] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [ ] Ported over testing from the old UI to new flow. Check to see there aren't any glaring errors here
- [ ] Added a few mocking Apollo calls, check to see there isn't anything glaring here

### Changes are Backwards Compatible
- [ ] Database changes are submitted as a separate PR
  - [ ] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [ ] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [ ] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] DevOps team has been notified if PR requires ops support
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

**Review by emmastephenson — COMMENTED** [review]
Overall looks great, thanks for your hard work on this! Just one addressable comment on removing the .env local file.

File: frontend/src/app/supportAdmin/PendingOrganizations/PendingOrganizations.tsx [hunk 1]
```diff
@@ -60,10 +56,50 @@ const PendingOrganizations = ({
...
+    let awaitingVerification = true;
+    try {
+      const valuesChangedArray = [
+        org.name === orgToVerify.name,
+        org.adminEmail === orgToVerify.adminEmail,
+        org.adminFirstName === orgToVerify.adminFirstName,
+        org.adminLastName === orgToVerify.adminLastName,
+        org.adminPhone === orgToVerify.adminPhone,
+      ];
+      const anyValueDifferent = !valuesChangedArray.every((v) => v === true);
```
Comment (nickclyde) [inline frontend/src/app/supportAdmin/PendingOrganizations/PendingOrganizations.tsx:77]: Nice!

File: frontend/src/app/supportAdmin/PendingOrganizations/PendingOrganizations.tsx [hunk 2]
```diff
@@ -60,10 +56,50 @@ const PendingOrganizations = ({
...
+        externalIdToVerify = updatedOrgExternalId;
+      }
+      submitIdentityVerified(externalIdToVerify);
+    } catch (e) {
+      console.error(e);
+    }
+
+    refetch();
+    if (awaitingVerification) {
+      console.log("nulling org to verify");
```
Comment (nickclyde) [inline frontend/src/app/supportAdmin/PendingOrganizations/PendingOrganizations.tsx:100]: Can you remove this stray console.log?

File: frontend/src/app/supportAdmin/PendingOrganizations/PendingOrganizationsContainer.test.tsx [hunk 3]
```diff
@@ -217,6 +249,63 @@ describe("PendingOrganizationsContainer", () => {
...
+      it("Space Camp submitted", async () => {
+        userEvent.click(
+          Array.from(await screen.findAllByText("View details"))[1]
+        );
+        userEvent.click(screen.getByText("Submit"));
+        expect(
+          await screen.findByText("Space Camp", { exact: false })
+        ).not.toBeInTheDocument();
+        expect(
+          await screen.findByText("A Real Hospital", { exact: false })
```
Comment (nickclyde) [inline frontend/src/app/supportAdmin/PendingOrganizations/PendingOrganizationsContainer.test.tsx:307]: Thanks for these tests! :100:

**Review by nickclyde — COMMENTED** [review]
This looks great! I'm a little confused by the texts of the buttons though. It's weird that "View details" actually brings up a modal to Edit/Verify. Maybe instead we could leave the header of that column blank and make the buttons say "Edit or verify identity" instead? And in the modal, it's not super clear to me which of the buttons will do which action. I'm thinking "Save details" should be "Update without verifying", and "Submit" should be "Verify". I'm not a designer/content person though - @Rebecca-LM or @vz3 what do you think?

**Comment by Rebecca-LM** [comment]
> This looks great! I'm a little confused by the texts of the buttons though. It's weird that "View details" actually brings up a modal to Edit/Verify. Maybe instead we could leave the header of that column blank and make the buttons say "Edit or verify identity" instead? And in the modal, it's not super clear to me which of the buttons will do which action. I'm thinking "Save details" should be "Update without verifying", and "Submit" should be "Verify". I'm not a designer/content person though - @Rebecca-LM or @vz3 what do you think?

@nickclyde thanks for flagging this stuff. We were working on this a bit with Bob and I didn't realize it was already going to production. I'd hoped to have another look before then!

I agree with removing the column header, then making the button "Edit/Verify". The button could be "Update only" and the other "Verify". If you could do a video or screenshot after it's always most helpful to see it in context to see if it works. Thanks!

**Comment by Rebecca-LM** [comment]
@fzhao99 did you need my review on this? I can't see any changes on my end rn.

**Review by nickclyde — APPROVED** [review]
Thanks for making those copy changes, great work on this!

**Review by emmastephenson — APPROVED** [review]
LGTM!! Can you notify the support slack once this goes in so we can alert the support folks of the change?

**Comment by zdeveloper** [comment]
This looks great, would've loved to remove `WIP` from the PR name, since that's the commit message that ended up on main.

**Pull request merged**